### PR TITLE
fix: [CDS-87172]: fixed Corrupted YAMLs being stored on creation flow of service / environment / input sets

### PR DESCRIPTION
**Pull request by karansaraswat19**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/10-common/components/RepoBranchSelectV2/RepoBranchSelectV2.tsx [hunk 1]
```diff
@@ -81,23 +81,25 @@ const hasToRefetchBranches = (
...
   repoName?: string
-) => {
+): boolean => {
   const responseMessages = (error?.data as Error)?.responseMessages
   return (
     !disabled &&
-    (gitProvider === Connectors.Harness || connectorIdentifierRef) &&
-    repoName &&
-    ((responseMessages?.length && responseMessages?.length > 0) || !!error)
+    Boolean(
```
Comment (chetanvekariya-harness) [inline src/modules/10-common/components/RepoBranchSelectV2/RepoBranchSelectV2.tsx:98]: nit: this needs to be simplified using independent statements.

File: src/modules/40-gitsync/components/GitSyncForm/GitSyncForm.tsx [hunk 2]
```diff
@@ -77,6 +77,7 @@ interface GitSyncFormProps<T> {
   skipDefaultConnectorSetting?: boolean
   skipBranch?: boolean
   supportNewBranch?: boolean
+  renderRepositoryLocationCard?: boolean
```
Comment (chetanvekariya-harness) [inline src/modules/40-gitsync/components/GitSyncForm/GitSyncForm.tsx:80]: nit: If it is boolean then, better name would have been `shouldRenderRepositoryLocationCard`

File: src/modules/40-gitsync/components/GitSyncForm/GitSyncForm.tsx [hunk 3]
```diff
@@ -152,6 +153,7 @@ function NewGitBranch<T extends GitSyncFormFields = GitSyncFormFields>(
         {!isNewBranch && (
           <RepoBranchSelectV2
             key={formikProps?.values?.repo}
+            gitProvider={formikProps.values.provider?.type}
```
Comment (chetanvekariya-harness) [inline src/modules/40-gitsync/components/GitSyncForm/GitSyncForm.tsx:156]: nit: `gitProviderType` would have been better prop name

File: src/modules/40-gitsync/components/GitSyncForm/__tests__/GitSyncForm.test.tsx [hunk 4]
```diff
@@ -135,7 +135,7 @@ describe('GitSyncForm test', () => {
         >
           {formikProps => (
             <FormikForm>
-              <GitSyncForm formikProps={formikProps} isEdit={false} />
+              <GitSyncForm formikProps={formikProps} isEdit={false} renderRepositoryLocationCard />
```
Comment (chetanvekariya-harness) [inline src/modules/40-gitsync/components/GitSyncForm/__tests__/GitSyncForm.test.tsx:138]: Have we added any test cases to test functionality of location cards?

File: src/modules/70-pipeline/components/InputSetForm/useSaveInputSet.ts [hunk 5]
```diff
@@ -106,6 +106,7 @@ export function useSaveInputSet(inputSetInfo: InputSetInfo): UseSaveInputSetRetu
       conflictCommitId?: string
     }): CreateUpdateInputSetsReturnType => {
       let response: ResponseInputSetResponse | undefined = undefined
+      const inputSetYaml = yamlStringify({ inputSet: clearNullUndefined(omit(inputSetObj, 'provider')) })
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/InputSetForm/useSaveInputSet.ts:109]: Why is this done here? Can we omit it where connectorRef, repoName etc are omitted?

**Review by chetanvekariya-harness — DISMISSED** [review]
Few comments that we should fix in develop. No need to fix in this HF PR.

**Review by sunnykesh-harness — APPROVED** [review]

**Pull request merged**